pkg/tsdb/elasticsearch/dataframe_builder.go

```diff
@@ -0,0 +1,149 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"sort"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+
+	es "github.com/grafana/grafana/pkg/tsdb/elasticsearch/client"
+)
+
+// processDocsToDataFrameFields converts documents to data frame fields
+func processDocsToDataFrameFields(docs []map[string]interface{}, propNames []string, configuredFields es.ConfiguredFields) []*data.Field {
+	size := len(docs)
+	isFilterable := true
+	allFields := make([]*data.Field, len(propNames))
+	timeString := ""
+	timeStringOk := false
+
+	for propNameIdx, propName := range propNames {
+		// Special handling for time field
+		if propName == configuredFields.TimeField {
+			timeVector := make([]*time.Time, size)
+			for i, doc := range docs {
+				// Check if time field is a string
+				timeString, timeStringOk = doc[configuredFields.TimeField].(string)
+				// If not, it might be an array with one time string
+				if !timeStringOk {
+					timeList, ok := doc[configuredFields.TimeField].([]interface{})
+					if !ok || len(timeList) != 1 {
+						continue
+					}
+					// Check if the first element is a string
+					timeString, timeStringOk = timeList[0].(string)
+					if !timeStringOk {
+						continue
+					}
+				}
+				timeValue, err := time.Parse(time.RFC3339Nano, timeString)
+				if err != nil {
+					// We skip time values that cannot be parsed
+					continue
+				} else {
+					timeVector[i] = &timeValue
+				}
```

Comment:
```suggestion
                if err != nil {
                    // We skip time values that cannot be parsed
                    continue
                } else {
                    tt := new(time.Time)
                    *tt = timeValue
                    timeVector[i] = tt
                }
```

We shouldn't take `&timeValue` (same address each loop). Allocate a fresh pointer.